lib/safe_macros.c

```diff
@@ -210,7 +210,12 @@ int safe_open(const char *file, const int lineno, void (*cleanup_fn) (void),
 	mode_t mode;
 
 	va_start(ap, oflags);
-	mode = va_arg(ap, mode_t);
+
+	if (sizeof(mode_t) < sizeof(int))
+		mode = va_arg(ap, int);
+	else
+		mode = va_arg(ap, mode_t);
+
 	va_end(ap);
 
 	rval = open(pathname, oflags, mode);
```

Comment:
That sounds fine, I should have added a comment along with the condition, if you're simplifying it can you add it for me? I'll do both this week otherwise.